amazon-kinesis-client-multilang/src/main/java/software/amazon/kinesis/multilang/config/KinesisClientLibConfigurator.java

```diff
@@ -69,8 +71,19 @@ public MultiLangDaemonConfiguration getConfiguration(Properties properties) {
         });
 
         Validate.notBlank(configuration.getApplicationName(), "Application name is required");
-        Validate.notBlank(configuration.getStreamName(), "Stream name is required");
+
+        if (configuration.getStreamArn() != null && !configuration.getStreamArn().trim().isEmpty()) {
+            final Arn streamArnObj = Arn.fromString(configuration.getStreamArn());
+            StreamIdentifier.validateArn(streamArnObj);
+            //Parse out the stream Name from the Arn (and/or override existing value for Stream Name)
+            final String streamNameFromArn = streamArnObj.resource().resource();
+            configuration.setStreamName(streamNameFromArn);
+
+        }
+
+        Validate.notBlank(configuration.getStreamName(), "Stream name or Stream Arn is required. Stream Arn takes precedence if both are passed in.");
```

Comment:
+1 for attempt at signaling "Stream Arn takes precedence ..."

_However_, this validation message only surfaces if the stream name is blank. Perhaps there should be a `log.warn(...)` -- circa ~79, before `configuration.setStreamName(...)` -- that warns of the name-stomping if-and-only-if a provided stream name is about to be overridden by the stream from the ARN.